torch/jit/_isinstance.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Dict, Tuple, Union, Optional
+
+
+def get_origin(target_type):
+    return getattr(target_type, "__origin__", None)
+
+
+def get_args(target_type):
+    return getattr(target_type, "__args__", None)
+
+
+def check_args_exist(target_type):
+    if target_type is List or target_type is list:
+        raise RuntimeError(
+            "Attempted to use List without a "
+            "contained type. Please add a contained type, e.g. "
+            "List[int]"
+        )
+    elif target_type is Tuple or target_type is tuple:
+        raise RuntimeError(
+            "Attempted to use Tuple without a "
+            "contained type. Please add a contained type, e.g. "
+            "Tuple[int]"
+        )
+    elif target_type is Dict or target_type is dict:
+        raise RuntimeError(
+            "Attempted to use Dict without "
+            "contained types. Please add contained type, e.g. "
+            "Dict[int, int]"
+        )
+    elif target_type is None or target_type is Optional:
+        raise RuntimeError(
+            "Attempted to use Optional without a "
+            "contained type. Please add a contained type, e.g. "
+            "Optional[int]"
+        )
+
+# supports List/Dict/Tuple and Optional types
+# TODO support future
+def generics_checker(obj, target_type):
+    origin_type = get_origin(target_type)
+    check_args_exist(target_type)
+    if origin_type is None:
+        pass
+    elif origin_type is list or origin_type is List:
```

Comment:
Out of curiosity, under what circumstances can `origin_type` be `typing.List`?

Reply:
In python 3.6 it's typing.List